.travis.yml

```diff
@@ -1,39 +1,42 @@
+  - HEROKU_APP=khalidashraf-todo-app
+  - secure: j9yGr8VrXd5o6pr6yAzAnK7DzyaDgZ8uKf5sp8tWKhu3wEuQenx3jFKndrl6w82AP3J+srBVw+yDv4e5GwGSSZ6HtF4F6AEUp+MVjg90NJwEjJuHO3FwsdkZNGncIEix044M7oA2nmoWQPZoSZ6XQRGZRa6xJrA4Y04s2Fwk+q9EdFnX4cRCcDc+7V9zkgdTEJneOEOpR2ElMz0z4/NWb357jmK37GdY4nvT3C+gEUqhKGp/de1FoNSLnlWmoKFXEtbUNqI4jhepf+QO4+KFlSJJBYOOPD308dt12vAdthv2VxKNYWh62KF4IfL/HAhPY13ZmWYtZC/K4aT1geoLkL1RHRIrvJQcCdSwQ+KwHRRsLE24SY1KeLxDOYCk5VZZ0wjjI1efeRjQc/MoGXSI0ems9bI5aWBQn2EVymHpRg8k+eMCcb2NUa8QU409hE73z0g0NvyIZRHzYS1CCZuEsbrXGcP0EOn8SD0+6mTo8PyqFzQ226t33/XEkbbU1ZGOHb7/eZKh3vU/iWx1AP4hakD9OF4dg3/2qgnnFItAkW5ZEu01O7AeHYr4As3n7mQqXVboq0Sb/eR3CC7JnZ4QoT30nc/5SBoYZYO9QkTW/Sh8e1bd+vVaAboOIF5S1+0nSTIy3mKQLsx7xygvLEGeaDsioL1JqWen7FKh68DD0Js=
+  - secure: Q+jfk7gIgAfkhoEEomitPwvh/PpOWG5pKgM08DX4DjvP+aSoc/nk9ijZnPzAwxZaoGKzb9M6MLNtRq4HSNLntCO55WWL2H43GlpYSuifZfGhg+684gx0oraz0H/QzppSEjayWNswEq6PVGp3TmVjSbfaDgowFW/VzPhJnS9dSUnlD0odStwGJTSMyFSGp2jWAZ8PoX7x5rf/Op1KhQXBHyP6dwL7B9tZkw+wu3mYb0fsvWONUHEJbjEiBg1gL5esrH6IIEU7F3KxN4WeYIDC3GTVwWjlolVo7pBds3YeFOX8shjlXWsaNam9BKDrkMuDKBYYzFzoZoAq8zXN6DbTKPKKYY4GotxStxuyh/Pzo6GjDFSi/X/b8R5y3249R8XkO1BDYo+7tFbnFI/q8kBvGx937cmwxQcKKxDUctIGZ/frLcR7NXtHHKOIni5hX6YWhRXH8EzJ2qYYMRNP2oRqnXNZ6HcolMeZgPYePLU39jK5sxzf+DApM3EJMDgjwANUCqwc1bx7gaR3ImMI2bf5+7uvXQXOLaW5QlFTYYDLmxVm+HR66h1/ObSa2gl8dIzjA4y6jg3mh8eC31ApEfvPYe9/6JJxmJ8P9WxvHCcTQNrwIu3FDhC06+NF9ZsLDL8W142TV9MmqdO9PEwFyfqPRxRxZzv9fGjhwHAwgFJ1KP0=
+  - OAUTH_AUTHENTICATE_URL=https://github.com/login/oauth/authorize
+  - OAUTH_ACCESS_TOKEN_URL=https://github.com/login/oauth/access_token
+  - OAUTH_USER_URL=https://api.github.com/user
+  - ROLE_WRITER_USERS=luckykhalid
+  - secure: E7arcKb4m0YGHIq6m8lgtyWQidsG6r42J2XEyomBylLk54K9Yz6b/GeErzj+RCtloomu958dDuweU4k92vLKF5TZmWAVpTcE7PP0Up2eJVouRdCx19tfSKlB7bXUuX93OyAZcwDNibTlkOem3lP8nxoEBZX+xLjv+X+asIpOWXLz/nLJia71Ay/SLVZyAG5slQR+QVmEC/+WAl7GNKKevADWkkqkOQbv6xnzEe7zntfplnJ55Hsyk9HwbjlSMXoearDMSjJio0CZ16gqK+gakSlsI85y1EMyg3l/3o5SDXtggGbmqX25ydOYzZZJMto82TqnamP7oF8/KN6aI76muaTM51Fe9dLqD+BWdAG34ig8tsviEE5ZTZte6lD31xB52jOQjUbWZEuiNPjyQ4NfGm778b4qJVdgmKE9G7Hgh5Tw0d3SWg7K6BTeTzH4ZuS0AzQabdnI3INecPGF+P5JFUijG94jLgtuErNRKJvyaWLRLrz7Pt9AR1XCqSuAZ1DsxITD4CyWxddq7uczPwrVzZAmuH2SCwK7U0zFYf+4OEXqXAgg0ukyah8S8WuNwMkDER386XhcrwtMpcYR0iaA6tCIlzNqQIywqNhqzN1ChC2zJFkRARVoYtBvWfKkZLxUVXBb/j5dQK6dHvqBbGuMGDWK0nHJ4+1HVuoFuekfi0A=  
+services:
+- docker
+  
+script:  
+  # Build test image and test the code  
+  - docker build --target test --tag todo_app:test .
+  - docker run --env-file .env.test todo_app:test tests
+  - docker run -e SECRET_KEY -e MONGO_USER -e MONGO_PASS -e OAUTH_CLIENT_ID -e OAUTH_CLIENT_SECRET -e OAUTH_AUTHENTICATE_URL -e OAUTH_ACCESS_TOKEN_URL -e OAUTH_USER_URL -e ROLE_WRITER_USERS todo_app:test tests_e2e
+  # Redeploy to Azure
+  - bash ./webhook.sh

```

Comment:
This should be part of your build-deploy.sh. You're currently updating your app service whenever any branch gets updated which would result in a very unstable site. The exercise (for M8) was to only deploy when the main branch is updated.

Reply:
Thank you @hugh-emerson  Moved to "bash ./webhook.sh" to build-deploy.sh file.